src/pages/SingleVideo/SingleVideo.jsx

```diff
@@ -0,0 +1,65 @@
+import "../../css/main.css";
+import "./SingleVideo.css";
+import { useState, useEffect } from "react";
+import { useParams } from "react-router-dom";
+import {
+  CardVideo,
+  CardLoader,
+  SingleVideoLoader,
+  SingleCardVideo,
+} from "../../components";
+import { useSingleVideo, useVideo } from "../../context";
+import { useDocumentTitle, useScrollToTop } from "../../utils";
+
+export const SingleVideo = () => {
+  useDocumentTitle("Easy TV | SingleVideo");
+  useScrollToTop();
+
+  const { videoId } = useParams();
+  const { getSingleVideo, singleVideo, issinglecardLoading, singlecardError } =
+    useSingleVideo();
+  const { getAllVideos, allVideos, iscardLoading, cardError } = useVideo();
+
+  const [filteredVideos, setFilteredVideos] = useState([]);
+
+  useEffect(() => {
+    getSingleVideo(videoId);
+    getAllVideos();
+  }, [videoId]);
+
+  useEffect(() => {
+    let tempVideos = [...allVideos];
+    tempVideos = tempVideos.filter(
```

Comment:
Instead of doing like this, can't you do something like
```javascript
const tempVideos = [...allVideos].filter()
```

Reply:
Yes, will change this. Thanks for pointing this out.

Reply:
yes will change it . Thanks for pointing this out